pylxd/image.py

```diff
@@ -80,7 +81,66 @@ def create(cls, client, image_data, public=False, wait=False):
             Operation.wait_for_operation(client, response.json()['operation'])
         return cls(client, fingerprint=fingerprint)
 
+    @classmethod
+    def create_from_config(cls, client, config, wait=False):
+        """ Create an image from the given configuration.
+
+        See: https://github.com/lxc/lxd/blob/master/doc/rest-api.md#post-6
+        """
+        if 'source' not in config or 'fingerprint' not in config['source']:
+            raise KeyError(
+                'You need to provide a fingerprint '
+                'in the config[\'source\'] argument'
+            )
+
+        response = client.api.images.post(json=config)
+        if wait:
+            Operation.wait_for_operation(client, response.json()['operation'])
+        return cls(client, fingerprint=config['source']['fingerprint'])
+
     def export(self):
         """Export the image."""
         response = self.api.export.get()
         return response.content
+
+    def copy_to_lxd(self, new_client, public=None, auto_update=None,
+                    wait=False):
+        """ Copy an image to a another LXD.
+
+        Destination host information is contained in the client
+        connection passed in.
+        """
+        self.sync()  # Make sure the object isn't stale
+
+        # TODO: this is a hacky way to remove the /1.0 do we have a better one?
+        url = '/'.join(self.client.api._api_endpoint.split('/')[:-1])
+
+        if public is None:
+            public = self.public
+
+        if auto_update is None:
+            auto_update = self.auto_update
+
+        config = {
```

Comment:
I like this pattern here. This is what I mean by not asking the user to create a random dict. We can construct this with our knowledge of LXD based on there parameters, and then the user doesn't need to know everything about LXD.